python_ta/checkers/invalid_name_checker.py

```diff
@@ -0,0 +1,525 @@
+                ):
+                    assigner = assign_type.value.elts[node.parent.elts.index(node)]
+                    if self._assigns_typevar(assigner):
+                        self._check_name(
+                            "typevar",
+                            assign_type.targets[0].elts[node.parent.elts.index(node)].name,
+                            node,
+                        )
+                        return
+                    if self._assigns_typealias(assigner):
+                        self._check_name(
+                            "typealias",
+                            assign_type.targets[0].elts[node.parent.elts.index(node)].name,
+                            node,
+                        )
+                        return
+
+                # Check classes (TypeVar's are classes so they need to be excluded first)
+                elif isinstance(inferred_assign_type, nodes.ClassDef):
+                    self._check_name("class", node.name, node)
+
+                # Don't emit if the name redefines an import in an ImportError except handler.
+                elif not _redefines_import(node) and isinstance(inferred_assign_type, nodes.Const):
+                    self._check_name("const", node.name, node)
+                else:
+                    self._check_name("variable", node.name, node)
+
+            # Check names defined in AnnAssign nodes
+            elif isinstance(assign_type, nodes.AnnAssign):
+                if utils.is_assign_name_annotated_with(node, "Final"):
+                    self._check_name("const", node.name, node)
+                elif self._assigns_typealias(assign_type.annotation):
+                    self._check_name("typealias", node.name, node)
+
+        # Check names defined in function scopes
+        elif isinstance(frame, nodes.FunctionDef):
+            # global introduced variable aren't in the function locals
+            if node.name in frame and node.name not in frame.argnames():
+                if not _redefines_import(node):
+                    self._check_name("variable", node.name, node)
+
+        # Check names defined in class scopes
+        elif isinstance(frame, nodes.ClassDef):
+            if not list(frame.local_attr_ancestors(node.name)):
+                for ancestor in frame.ancestors():
+                    if utils.is_enum(ancestor) or utils.is_assign_name_annotated_with(
+                        node, "Final"
+                    ):
+                        self._check_name("class_const", node.name, node)
+                        break
+                else:
+                    self._check_name("class_attribute", node.name, node)
+
+    def _check_name(self, node_type: str, name: str, node: nodes.NodeNG) -> None:
+        """Check for a name that violates the PEP8 naming conventions."""
+        name_check = NAME_CHECK[node_type]
+        # Update node_type to be more readable
```

Comment:
It isn't necessary to do any string parsing, we don't need to follow what pylint does. Just pick the (human-friendly) name that you want for each node type and use it directly.